### PR TITLE
Fix NoAMethodError in FaradayServerError

### PR DESCRIPTION
**Pull request by tomerpar133**
Problem:
Error when Faraday::ClientError.new(nil)

Solution
added try to the response hash on FaradayServerError

### CODE REVIEW

**Comment by dblock** [comment]
For rubocop, `rubocop -a ; rubocop --auto-gen-config`.

Needs tests.

Needs changelog.

Thanks.

File: lib/graphlient/errors/faraday_server_error.rb [hunk 1]
```diff
@@ -4,8 +4,8 @@ class FaradayServerError < ServerError
       def initialize(inner_exception)
         super(inner_exception.message, inner_exception)
         @inner_exception = inner_exception
-        @response = inner_exception.response[:body]
-        @status_code = inner_exception.response[:status]
+        @response = inner_exception.response.try(:[],:body)
+        @status_code = inner_exception.response.try(:[],:status)
```
Comment (yuki24) [inline lib/graphlient/errors/faraday_server_error.rb]: The `#try` method is a method that[ ActiveSupport defines](https://api.rubyonrails.org/classes/Object.html#method-i-try) on almost all objects and we don't want to depend on it.

**Comment by dblock** [comment]
https://github.com/ashkan18/graphlient/issues/51

**Comment by neroleung** [comment]
I do not agree with the original author on certain things so I decided to keep working on my own fork until the main branch is fixed. But in the meantime if any of you really need a fix on this soon, check out https://github.com/neroleung/graphlient.

**Comment by dblock** [comment]
> I do not agree with the original author on certain things so I decided to keep working on my own fork until the main branch is fixed. But in the meantime if any of you really need a fix on this soon, check out https://github.com/neroleung/graphlient.

What are these things you don't agree with @neroleung. This library is written and maintained by a bunch of people, we're happy to discuss anything and help you contribute to the parent repo rather than fork away.

**Comment by neroleung** [comment]
My apologies. My project has a tight deadline and whether this bug can be fixed quickly or not determines if I need to drop this gem in my project. I didn't expect quick responses and reviews like you all did in open source repos. So I got anxious. I'll make new PRs and fix things up in stages like you all suggested. Thanks~

**Comment by dblock** [comment]
Closed via https://github.com/ashkan18/graphlient/pull/68.

**Pull request closed without merging**